core/actions/data_preparation.ts

```diff
@@ -34,19 +34,25 @@ export class DataPreparation extends ActionBuilder<dataform.DataPreparation> {
 
     config.filename = resolveActionsConfigFilename(config.filename, configPath);
     const dataPreparationAsJson = nativeRequire(config.filename).asJson;
-    const dataPreparationDefinition = parseDataPreparationDefinitionJson(dataPreparationAsJson);
+    const dataPreparationDefinition = this.parseDataPreparationDefinitionJson(dataPreparationAsJson);
 
-    // Find targets
-    const targets = getTargets(dataPreparationDefinition);
-    this.proto.targets = targets.map(target =>
-      this.applySessionToTarget(target, session.projectConfig, config.filename, true)
-    );
-    this.proto.canonicalTargets = targets.map(target =>
-      this.applySessionToTarget(target, session.canonicalProjectConfig)
-    );
+    const targets = this.getTargets(dataPreparationDefinition);
+    this.proto.targets = targets
+        .map(target =>
+            this.applySessionToTarget(target, session.projectConfig, config.filename, true)
+        ).map(target =>
+            this.applyDataPreparationDefaultsToTarget(dataPreparationDefinition, target)
+        );
```

Comment:
I think this order should be reversed, against your PR description.

I think defaults should be applied at the level of granularity:

Workflow Setting defaults > Data Preparation defaults > explicit table targets

This way you can have two different data preparations in the same project have different default values within themselves, rather than always having the workflow settings override them